Cubical/Algebra/CommRing/Quotient/IdealSum.agda

```diff
@@ -0,0 +1,170 @@
+open import Cubical.Algebra.CommRing.Ideal.Base
+open import Cubical.Algebra.CommRing.Ideal.Sum
+open import Cubical.Algebra.CommRing.Ideal.SurjectiveImage
+
+open import Cubical.Tactics.CommRingSolver.Reflection
+
+private variable ℓ : Level
+
+module Construction {R : CommRing ℓ} (I J : IdealsIn R) where
+  open CommIdeal ⦃...⦄
+  open IdealSum R
+  open Cubical.Algebra.Ring.Properties.RingHoms
+  open CommRingStr ⦃...⦄
+  open IsRingHom ⦃...⦄
+
+  {-
+    We show ψ is an isomorphism, by showing it is
+    surjectiv and has a trivial kernel.
+
+              R ──π₁─→ R/I
+              |         |
+              |         π₂
+              ↓         ↓
+            R/I+J ─ψ→ (R/I)/J
+
+  -}
+
+  π₁ = quotientHom R I
+
+  π₁J : IdealsIn (R / I)
+  π₁J = imageIdeal π₁ (quotientHomSurjective R I) J
+
+  π₂ = quotientHom (R / I) π₁J
+
+  π : CommRingHom R ((R / I) / π₁J)
+  π = π₂ ∘r π₁
+
+  π+ : CommRingHom R (R / (I +i J))
+  π+ = quotientHom R (I +i J)
+
+  private instance _ = snd R
+                   _ = snd (R / I)
+                   _ = snd ((R / I) / π₁J)
+                   _ = snd (R / (I +i J))
+                   _ = R
+                   _ = (R / I)
+                   _ = ((R / I) / π₁J)
+
+  πI+J≡0 : (x : ⟨ R ⟩) → x ∈ (I +i J) → fst π x ≡ 0r
+  πI+J≡0 x  =
+    PT.rec (is-set _ _)
+           λ ((a , b) , (a∈I , b∈J , x≡a+b))
+             → let instance _ = snd π
+                   πa≡0 : fst π a ≡ 0r
+                   πa≡0 = fst π₂ (fst π₁ a) ≡⟨ cong (fst π₂) (zeroOnIdeal I a a∈I) ⟩
+                          fst π₂ 0r         ≡⟨ pres0 ⦃ snd π₂ ⦄ ⟩
+                          0r ∎
+                   πb≡0 : fst π b ≡ 0r
+                   πb≡0 = zeroOnIdeal π₁J (fst π₁ b) ∣ b , (b∈J , refl) ∣₁
+               in fst π x           ≡⟨ cong (fst π) x≡a+b ⟩
+                  fst π (a + b)     ≡⟨ pres+ a b ⟩
+                  fst π a + fst π b ≡[ i ]⟨ πa≡0 i + πb≡0 i ⟩
+                  0r + 0r           ≡⟨ +IdL _ ⟩
+                  0r ∎
+
+  π⁻¹0≡I+J : (x : ⟨ R ⟩) → fst π x ≡ 0r → x ∈ (I +i J)
+  π⁻¹0≡I+J x πx≡0 = PT.rec isPropPropTrunc (λ (b , b∈J , π₁b≡π₁x) → step2 b b∈J π₁b≡π₁x) π₁x∈J
+    where π₁x∈J : (fst π₁ x) ∈ π₁J
+          π₁x∈J = subst (fst π₁ x ∈_) (kernel≡I π₁J) πx≡0
+
+          step1 : ∀ b → fst π₁ b ≡ fst π₁ x → - (b - x) ∈ I
+          step1 b π₁b≡π₁x =
+            -Closed I (b - x)
+              (subst (λ K → b - x ∈ K) (kernel≡I I)
+                  (kernelFiber R (R / I) π₁ b x π₁b≡π₁x))
+
+          useSolver : (x b : ⟨ R ⟩) → x ≡ - (b - x) + b
+          useSolver = solve R
+
+          step2 : ∀ b → b ∈ J → fst π₁ b ≡ fst π₁ x → x ∈ (I +i J)
+          step2 b b∈J π₁b≡π₁x =
+            ∣ (- (b - x) , b) , (step1 b π₁b≡π₁x) , (b∈J , (x ≡⟨ useSolver x b ⟩ (- (b - x) + b) ∎)) ∣₁
+
+  ψ : CommRingHom (R / (I +i J)) ((R / I) / π₁J)
+  ψ = UniversalProperty.inducedHom R ((R / I) / π₁J) (I +i J) π πI+J≡0
+
+  kernel-0 : (x : ⟨ R / (I +i J) ⟩) → fst ψ x ≡ 0r → x ≡ 0r
```

Comment:
I think you could write this as "`kernelIdeal ψ` is contained in the zero ideal".

Reply:
Agreed offline to not do this...